magnum_capi_helm/driver.py

```diff
@@ -572,6 +572,34 @@ def _get_fixed_network_id(self, context, cluster):
                 context, network, source="name", target="id", external=False
             )
 
+    def _validate_allowed_flavor(self, context, requested_flavor):
+        # Compare requested flavor with allowed for Kubernetes node
+        flavors = (
+            clients.OpenStackClients(context)
+            .nova()
+            .flavors.list(min_ram=CONF.capi_helm.minimum_flavor_ram)
```

Comment:
I feel there must be a more efficient way, but this is less API requests than looking up by name and id, so maybe this is the best option.